packages/cli/internal/pkg/osutils/utils_test.go

```diff
@@ -0,0 +1,142 @@
+	_, err := DetermineHomeDir()
+
+	expectedError := actionableerror.New(err, "Please check that your home or user profile directory is defined within your environment variables")
+
+	assert.Error(t, err, expectedError)
+}
+
+func TestExpandHomeDir_WithExpansion(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockOs := iomocks.NewMockOS(ctrl)
+	osUserHomeDir = mockOs.UserHomeDir
+
+	expectedHomePath := "/home/user"
+	expectedExpandedPath := "/home/user/FooBar"
+
+	mockOs.EXPECT().UserHomeDir().Return(expectedHomePath, nil)
+	actualExpandedPath, _ := ExpandHomeDir("~/FooBar")
+
+	assert.Equal(t, expectedExpandedPath, actualExpandedPath)
+	ctrl.Finish()
+}
+
+func TestExpandHomeDir_WithExpansion_HomeDirOnly(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockOs := iomocks.NewMockOS(ctrl)
+	osUserHomeDir = mockOs.UserHomeDir
+
+	expectedHomePath := "/home/user"
+	expectedExpandedPath := "/home/user"
+
+	mockOs.EXPECT().UserHomeDir().Return(expectedHomePath, nil)
+	actualExpandedPath, _ := ExpandHomeDir("~")
+
+	assert.Equal(t, expectedExpandedPath, actualExpandedPath)
+	ctrl.Finish()
+}
+
+func TestExpandHomeDir_WithExpansion_HomeDirOnlyWithSlash(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockOs := iomocks.NewMockOS(ctrl)
+	osUserHomeDir = mockOs.UserHomeDir
+
+	expectedHomePath := "/home/user"
+	expectedExpandedPath := "/home/user"
+
+	mockOs.EXPECT().UserHomeDir().Return(expectedHomePath, nil)
+	actualExpandedPath, _ := ExpandHomeDir("~/")
+
+	assert.Equal(t, expectedExpandedPath, actualExpandedPath)
+	ctrl.Finish()
+}
+
+func TestExpandHomeDir_WithExpansionError(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockOs := iomocks.NewMockOS(ctrl)
+	osUserHomeDir = mockOs.UserHomeDir
+	expectedOsError := errors.New("OS Error")
+	mockOs.EXPECT().UserHomeDir().Return("", expectedOsError)
+	_, err := ExpandHomeDir("~/Error/Dir")
+
+	expectedError := actionableerror.New(err, "Please check that your home or user profile directory is defined within your environment variables")
+
+	assert.Error(t, err, expectedError)
+	ctrl.Finish()
+}
+
+func TestExpandHomeDir_WithoutExpansion(t *testing.T) {
+	tests := map[string]struct {
+		pathTobeExpanded string
+		expectedPath     string
+	}{
+		"Tilde at the beginning": {
+			pathTobeExpanded: "~~~",
+			expectedPath:     "~~~",
+		},
+		"Tilde at the beginning with escape and backslash": {
+			pathTobeExpanded: "\\~\\~\\~/",
+			expectedPath:     "\\~\\~\\~/",
+		},
+		"Tilde at the beginning with escape": {
+			pathTobeExpanded: "~\\~\\~",
+			expectedPath:     "~\\~\\~",
+		},
+		"Tilde in the middle": {
+			pathTobeExpanded: "Foo~/~Bar",
+			expectedPath:     "Foo~/~Bar",
+		},
+		"Tilde in the end": {
+			pathTobeExpanded: "~Foo/Bar~",
+			expectedPath:     "~Foo/Bar~",
+		},
+		"Empty string": {
+			pathTobeExpanded: "",
+			expectedPath:     "",
+		},
+		"Relative": {
+			pathTobeExpanded: "FooBar",
+			expectedPath:     "FooBar",
+		},
+		"Absolute": {
```

Comment:
I don't see any test cases that actually expand a path. for all test cases pathTobeExpanded == expectedPath. Am I missing anything?

Reply:
Yeah these ones are not supposed to be expanded as it's not `~` nor `~/...`. I'll change the variable name to input so it's less confusing. The ones to be expanded are tested in unit tests between line 38 and 81.

Reply:
Synced offline. Will move all positive cases into a parameterized test.